internal/karma/ts_web_test.bzl

```diff
@@ -78,6 +78,15 @@ def _ts_web_test_impl(ctx):
         for f in ctx.files.static_files
     ]
 
+    setup_files = []
+    for d in ctx.attr.setup_deps:
+        if not hasattr(d, "typescript"):
+            fail("labels in setup_deps must be created by ts_library")            
+        es5_sources = d.typescript.es5_sources.to_list()
+        if len(es5_sources) != 1:
+            fail("labels in setup_deps must contain exactly one file")   
```

Comment:
it may be better to allow multiple files per label as not all setup steps have to be ordered so it would be less overhead for a user with multiple setup steps to be able to use a single ts_library with a glob

Reply:
I'm fine either way, it just depends on what's the most idiomatic approach in Bazel. 

In the more permissive scenario (any number of files per label) it's up to the user to not shoot themselves in the foot since they are all effectively entry points and the order of execution is undefined.

I'm not sure there are a lot of cases beyond super trivial ones where the order doesn't matter though. I guess we can chalk it up to user fault if they have a bunch of modules that do stuff at the toplevel.

Reply:
Good question about what is more idiomatic. I'm in favor of allowing multiple inputs and documenting that the order of execution of the files within each label is not deterministic. @alexeagle What do you think?

Reply:
Looking at this with @kyliau 

Maybe it would be better to use the same pattern from `nodejs_binary`.

1. list these extra deps as normal in the `deps`. they won't be executed since they don't match the filename filter for specs
2. the setup argument would be a list of strings, each of which is to be `require`d


"bootstrap | List of strings; Optional; Default is []JavaScript modules to be loaded before the entry point. For example, Angular uses this to patch the Jasmine async primitives for zone.js before the first describe."